src/firebolt/service/V2/database.py

```diff
@@ -33,9 +21,44 @@ class DatabaseService(BaseService):
         "attached_engine_name_contains",
     ]
 
+    @cached_property
+    def catalog_name(self) -> str:
+        query = "SELECT count(*) FROM information_schema.catalogs"
+        with self._connection.cursor() as c:
+            try:
+                c.execute(query)
+            except OperationalError:
+                return "database"
+            return "catalog"
+
+    @cached_property
+    def db_fields(self) -> Tuple[str, ...]:
+        return (
+            f"{self.catalog_name}_name",
+            "description",
+            "region",
+            "uncompressed_size",
+            "compressed_size",
+            "attached_engines",
+            "created_on",
+            "created_by",
+            "errors",
+        )
+
+    @cached_property
+    def get_sql(self) -> str:
+        return (
+            f"SELECT {', '.join(self.db_fields)} "
+            f"FROM information_schema.{self.catalog_name}s"
+        )
+
+    @cached_property
+    def get_by_name_sql(self) -> str:
+        return self.get_sql + " WHERE database_name=?"
```

Comment:
small point, but I think at this stage `@cached_property` is an overkill. All properties it calls are also cached